js/src/api/contract/contract.js

```diff
@@ -232,51 +233,102 @@ export default class Contract {
       return this._subscribe(event, options, callback);
     };
 
+    event.register = (options = {}, callback) => {
```

Comment:
The `register` method actually calls `eth_getFilterChanges` every second, which is unnecessary for getting token meta data updates (will be max a change for every block). We could actually call it every minute or so if needed...